### ISSUE TITLE
load.Instances with string literal CUE content

### ISSUE THREAD

**Issue by dghubble** [cue-lang/cue#3164]
**Is your feature request related to a problem? Please describe.**

I'd like to use the cue Go API to load cue content from both literal content and paths, but the API seems unclear if it supports this.

```cue
# foo.cue
{
  name: config.name
}
```

```go

instances := load.Instances("foo.cue", config)

literal := `
  config: {
    name: "bar"
  }
`

cuectx := cuecontext.New()
value1 := cuectx.CompileString(content)
value2, _ := cuectx.BuildInstances(instances)

// how to build literal and instances at the same time? or get the literal to be an "instance"?
// desired output
{
  name: "bar"
}
```

With literal CUE contents, it seems like about all the API let's us do is convert them directly to a `Value`. Whereas in this example, we want to "evaluate" them all together.  The `load.Instances(args, config)` takes string filenames, whereas most Go APIs would offer an `io.Reader` that would let use provide the content to be loaded in a variety of ways.

Overall, the terms context, instance, build, compile, and load make it very unclear how to use the Go API. I've gathered that to "build" cue stuff together you need to have all the bits of config be available. Values are for "fully rendered" stuff, you can't take the string content value and the BuildInstances Value and merge them together.

Is it possible to combine expressions that originate from string literals and files? Or am I required to write them out to files?

**Comment by dghubble** [comment]
tldr: I've got a bunch of CUE contents, some are in Go strings some are files on disk. How do I use this Go library?
